Assignment4/mihai-cristian3+07932683/task07.py

```diff
@@ -0,0 +1,130 @@
+# -*- coding: utf-8 -*-
+"""Task07.ipynb
+
+Automatically generated by Colaboratory.
+
+Original file is located at
+    https://colab.research.google.com/drive/1QBTpxFNvBZBPxWpOT11OQ7EEOMi7ZjcZ
+
+**Task 07: Querying RDF(s)**
+"""
+
+!pip install rdflib 
+github_storage = "https://raw.githubusercontent.com/FacultadInformatica-LinkedData/Curso2021-2022/master/Assignment4/course_materials"
+
+"""Leemos el fichero RDF de la forma que lo hemos venido haciendo"""
+
+from rdflib import Graph, Namespace, Literal
+from rdflib.namespace import RDF, RDFS
+g = Graph()
+g.namespace_manager.bind('ns', Namespace("http://somewhere#"), override=False)
+g.namespace_manager.bind('vcard', Namespace("http://www.w3.org/2001/vcard-rdf/3.0#"), override=False)
+g.parse(github_storage+"/rdf/example6.rdf", format="xml")
+
+"""**TASK 7.1: List all subclasses of "Person" with RDFLib and SPARQL**"""
+
+# RDFlib way
+from rdflib.namespace import RDF
+ns = Namespace("http://somewhere#")
+
+# Visualize the results
+print("RDFlib way")
+for result11 in g.triples((None, RDFS.subClassOf, ns.Person)):
+  print(result11)
+  for result12 in g.triples((None, RDFS.subClassOf, result11[0])):
+    print(result12)
+  
+
+
+# SPARQL way
+from rdflib.plugins.sparql import prepareQuery
+
+q1 = prepareQuery("""
+SELECT ?sub WHERE {
+  {
+    ?sub RDFS:subClassOf ns:Person
+  } UNION {
+    ?aux RDFS:subClassOf ns:Person .
+    ?sub RDFS:subClassOf ?aux
```

Comment:
this recovers 2 levels of the taxonomy. If you use * you will retrieve all of them